### ISSUE TITLE
Saving URL or attachment_id on file/image fields

### ISSUE THREAD

**Issue by tareq1988** [tareq1988/wordpress-settings-api-class#48]
Right now we are saving attachment URLs in the option. Wouldn't it be better to save attachment ids instead?

So if its a file, we can use `wp_get_attachment_url()`. If its an image, we could get `full`, `thumbnail`, `medium` images if we wanted. What do you guys think?

Changing to attachment_id will also break the existing forms as users are already fetching URLs and acting on that.

**Comment by roidayan** [comment]
I'm in favor for the id.

**Comment by roidayan** [comment]
but it also means it must use the media uploader. you won't be able to use external url.

**Comment by tareq1988** [comment]
For external url, you can use the text input anyway.

**Comment by roidayan** [comment]
yes so maybe i dont understand what the ticket is about. adding another field that will use attachment id?

**Comment by tareq1988** [comment]
No, the existing file field saves the URL, from now it'll save the attachment id. Right now you can copy paste external urls, but it doesn't have to be a `file` field, you could use a `text` field to save anything including an url.

**Comment by roidayan** [comment]
I use the file field so user can use the media uploader or use a link. i dont want a text field here.

**Comment by fredbradley** [comment]
Just by way of a plus one... saving the `attachment_id` would be far more useful than saving the full url. 

I want to allow users to upload one file (it's a default fallback in case contributors don't set a featured image on posts) because it's an IMAGE I want to be able to interact with it so I can get the thumb size, medium size, etc etc...
